VMEncryption/main/ResourceDiskUtil.py

```diff
@@ -0,0 +1,274 @@
+        return False
+
+    def is_luks_device(self):
+        """ checks if the device is set up with a luks header """
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'cryptsetup isLuks ' + self.RD_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def is_luks_device_opened(self):
+        """ check for presence of luks uuid to see if device was already opened """
+        # suppress logging to avoid log clutter if the device is not open yet
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'test -b /dev/disk/by-uuid/$(cryptsetup luksUUID ' + self.RD_DEV_PATH + ')'
+        return (int)(self.executor.ExecuteInBash(cmd, suppress_logging=True)) == 0
+
+    def is_valid_key(self):
+        """ test if current key can be used to open current partition """
+        # suppress logging to avoid log clutter if the key doesn't match
+        if not self.resource_disk_partition_exists():
+            return False
+        cmd = 'cryptsetup luksOpen ' + self.RD_DEV_PATH + ' --test-passphrase --key-file ' + self.RD_KEY_FILE
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def resource_disk_exists(self):
+        """ true if the udev name for resourced disk exists """
+        cmd = 'test -b ' + self.RD_BASE_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def resource_disk_partition_exists(self):
+        """ true if udev name for resource disk partition exists """
+        cmd = 'test -b ' + self.RD_DEV_PATH
+        return (int)(self.executor.Execute(cmd, suppress_logging=True)) == 0
+
+    def format_luks(self):
+        """ set up resource disk crypt device layer using disk util """
+        if not self.resource_disk_partition_exists():
+            self.logger.log('LUKS format operation requested, but resource disk partition does not exist')
+            return False
+        return (int)(self.disk_util.luks_format(passphrase_file=self.RD_KEY_FILE, dev_path=self.RD_DEV_PATH, header_file=None)) == 0
+
+    def encrypt(self):
+        """ use disk util with the appropriate device mapper """
+        self.mount_key_volume()
+        return (int)(self.disk_util.encrypt_disk(dev_path=self.RD_DEV_PATH, passphrase_file=self.RD_KEY_FILE, mapper_name=self.mapper_name, header_file=None)) == 0
+
+    def make(self):
+        """ make a default file system on top of the crypt layer """
+        make_result = self.disk_util.format_disk(dev_path=self.mapper_path, file_system=CommonVariables.default_file_system)
+        if make_result != CommonVariables.process_success:
+            self.logger.log(msg="Failed to make file system on ephemeral disk", level=CommonVariables.ErrorLevel)
+            return False
+        # todo - drop DATALOSS_WARNING_README.txt file to disk
+        return True
+
+    def mount_key_volume(self):
```

Comment:
the BekUtil.py already has this method (sort of).

Reply:
get_bek_passphrase_file did not seem to be a close enough fit for this method and I wanted to include the operation in test coverage